src/Features/CSharp/Portable/CodeRefactorings/InvertIf/InvertIfCodeRefactoringProvider.Analyzer.cs

```diff
@@ -0,0 +1,365 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.CodeRefactorings.InvertIf
+{
+    internal sealed partial class CSharpInvertIfCodeRefactoringProvider
+    {
+        private sealed class Analyzer : Analyzer<IfStatementSyntax>
+        {
+            protected override string GetTitle()
+            {
+                return CSharpFeaturesResources.Invert_if;
+            }
+
+            protected override SyntaxNode GetRootWithInvertIfStatement(
+                Document document,
+                SemanticModel semanticModel,
+                IfStatementSyntax ifStatement,
+                InvertIfStyle invertIfStyle,
+                SyntaxNode subsequenceSingleExitPointOpt,
+                CancellationToken cancellationToken)
+            {
+                var generator = SyntaxGenerator.GetGenerator(document);
+                var syntaxFacts = CSharpSyntaxFactsService.Instance;
+
+                var ifNode = ifStatement;
+
+                var negatedCondition = (ExpressionSyntax)Negator.Negate(ifStatement.Condition, generator, syntaxFacts, semanticModel, cancellationToken);
```

Comment:
nit: please wrap.  my litmus is: does this stretch *far* past what github can show.  it just makes it much easier to review.  thanks!

Reply:
(pending)